exchanges/gateio/gateio_websocket.go

```diff
@@ -764,6 +733,146 @@ func singleSymbolChannel(name string) bool {
 	return false
 }
 
+// ValidateSubscriptions implements subscription.SubscriptionsValidator interface
+func (g *Gateio) ValidateSubscriptions(l subscription.List) error {
+	orderbookGuard := map[key.PairAsset]string{}
+	for _, s := range l {
+		n := channelName(s)
+		if !isRestrictedOrderbookChannel(n) {
```

Comment:
🐒 `restricted` doesn't feel like the right adjective for this. `shared` or `clashing` maybe
```suggestion
		if !isSharedOrderbookChannel(n) {
```